### PR TITLE
[4.3][Guided Tours] Remove unnecessary title generation on copy or duplicate

### PR DESCRIPTION
**Pull request by obuisard**
Pull Request for Issue #40024.

### Summary of Changes

The tour and step titles get a number when duplicated. 
However, this is unnecessary, especially that it breaks translations if the title is a language key. 

### Testing Instructions

Go to the tours list view.
Edit a tour.
Save as Copy. 
You should end up with a disabled tour but matching the one it was copied from.

Select one or several tours. 
Go to 'Actions' and select 'Duplicate'.
You should end up with disabled tour(s) but matching the one(s) it(they) was(were) copied from.

Go to the step view of a tour (to access the step view, click on the number next to the tour in tours list view).
Edit a step.
Save as Copy.
You should end up with a disabled step but matching the one it was copied from.

### Actual result BEFORE applying this Pull Request

A tour title could end up looking like:
COM_GUIDEDTOURS_TOUR_USERS_TITLE (2)
in tours list view because it can no longer be translated (if the original title is a language key).

### Expected result AFTER applying this Pull Request

You get the proper translation for the tour if the title is a language key.
Titles no longer get an additional differentiable number.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_guidedtours/src/Model/StepModel.php [hunk 1]
```diff
@@ -67,29 +67,6 @@ protected function canDelete($record)
...
-     *
-     * @since   4.3.0
-     */
-    protected function generateNewTitle($categoryId, $alias, $title)
-    {
-        // Alter the title
-        $table = $this->getTable();
-
-        while ($table->load(['title' => $title])) {
-            $title = StringHelper::increment($title);
```
Comment (Quy) [inline administrator/components/com_guidedtours/src/Model/StepModel.php:87]: Please remove unused import statement.

Comment (obuisard) [reply administrator/components/com_guidedtours/src/Model/StepModel.php:87]: Done, thanks.

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on b6012a9db667873314328dd5c5514d22f611ae84<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40063">issues.joomla.org/tracker/joomla-cms/40063</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on b6012a9db667873314328dd5c5514d22f611ae84<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40063">issues.joomla.org/tracker/joomla-cms/40063</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40063">issues.joomla.org/tracker/joomla-cms/40063</a>.</sub>

**Pull request merged**